spec/abstractops.html

```diff
@@ -1281,60 +1246,56 @@ <h1>ParseTemporalDurationString ( _isoString_ )</h1>
-        1. Throw a *RangeError* exception.
-      1. Let _sign_, _years_, _months_, _weeks_, _days_, _hours_, _fHours_, _minutes_, _fMinutes_, _seconds_, and _fSeconds_ be the parts of _isoString_ produced respectively by the |Sign|, |DurationYears|, |DurationMonths|, |DurationWeeks|, |DurationDays|, |DurationWholeHours|, |DurationHoursFraction|, |DurationWholeMinutes|, |DurationMinutesFraction|, |DurationWholeSeconds|, and |DurationSecondsFraction| productions, or *undefined* if not present.
-      1. If _sign_ is the code unit 0x002D (HYPHEN-MINUS) or 0x2212 (MINUS SIGN), then
-        1. Let _factor_ be −1.
-      1. Else,
-        1. Let _factor_ be 1.
-      1. Set _years_ to ? ToIntegerOrInfinity(_years_) × _factor_.
-      1. Set _months_ to ? ToIntegerOrInfinity(_months_) × _factor_.
-      1. Set _weeks_ to ? ToIntegerOrInfinity(_weeks_) × _factor_.
-      1. Set _days_ to ? ToIntegerOrInfinity(_days_) × _factor_.
-      1. Set _hours_ to ? ToIntegerOrInfinity(_hours_) × _factor_.
-      1. Set _minutes_ to ? ToIntegerOrInfinity(_minutes_) × _factor_.
-      1. Set _seconds_ to ? ToIntegerOrInfinity(_seconds_) × _factor_.
-      1. If _fSeconds_ is not *undefined*, then
-        1. Let _fSecondsDigits_ be the substring of _fSeconds_ from 1.
-        1. Let _fSecondsDigitsExtended_ be the string-concatenation of _fSecondsDigits_ and *"000000000"*.
-        1. Let _milliseconds_ be the substring of _fSecondsDigitsExtended_ from 0 to 3.
-        1. Set _milliseconds_ to ! ToIntegerOrInfinity(_milliseconds_) × _factor_.
-        1. Let _microseconds_ be the substring of _fSecondsDigitsExtended_ from 3 to 6.
-        1. Set _microseconds_ to ! ToIntegerOrInfinity(_microseconds_) × _factor_.
-        1. Let _nanoseconds_ be the substring of _fSecondsDigitsExtended_ from 6 to 9.
-        1. Set _nanoseconds_ to ! ToIntegerOrInfinity(_nanoseconds_) × _factor_.
-      1. Else,
-        1. Let _milliseconds_ be 0.
-        1. Let _microseconds_ be 0.
-        1. Let _nanoseconds_ be 0.
-      1. If _fHours_ is not *undefined*, then
-        1. Let _fHoursDigits_ be the substring of _fHours_ from 1.
+      1. Let _duration_ be ParseText(_isoString_, |TemporalDurationString|).
+      1. If _duration_ is a List of errors, throw a *RangeError* exception.
+      1. Let each of _sign_, _years_, _months_, _weeks_, _days_, _hours_, _fHours_, _minutes_, _fMinutes_, _seconds_, and _fSeconds_ be the source text matched by the respective |Sign|, |DurationYears|, |DurationMonths|, |DurationWeeks|, |DurationDays|, |DurationWholeHours|, |DurationHoursFraction|, |DurationWholeMinutes|, |DurationMinutesFraction|, |DurationWholeSeconds|, and |DurationSecondsFraction| Parse Node enclosed by _duration_, or an empty sequence of code points if not present.
+      1. Let _yearsMV_ be ! ToIntegerOrInfinity(CodePointsToString(_years_)).
```

Comment:
```suggestion
      1. Let _yearsMV_ be ! ToIntegerOrInfinity(! CodePointsToString(_years_)).
```
(and same for the lines below?)

Reply:
I think the inner assertion of infallibility is superfluous in the presence of the outer one, but will make this change if you feel strongly about it.

Reply:
I don't feel strongly about it.